Add Longest Word in Coffeescript

stefanurkal opened the pull request:
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][renegade-coder]! 
For simplicity, please make sure that your pull request includes one and only one contribution.

Please fill _one_ of the sections below as applicable.
Please also add any other relevant information to the Notes section at the bottom.
You may delete or just ignore any other sections.
For more information please refer to our [contributing documentation][contributing]

## I Am Adding a New Code Snippet in an Existing Language

- [x] I fixed #4076
- [x] I named the pull request using `Add {PROJECT} in {LANGUAGE}` format

## Other Notes

I used the same codespace as previous attempt. 



[renegade-coder]: https://therenegadecoder.com/
[contributing-plagiarism]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism
[contributing-new-project]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#requirements-for-a-new-project
[contributing-new-language]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#requirements-for-a-new-language
[contributing-readme]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#create-readmes
[contributing-tests-in-detail]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#tests-in-detail
[contributing]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md
[sample-programs-project-list]: https://sampleprograms.io/projects/
[contributing-modifications]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#modifying-existing-code-snippets

rzuckerm commented:
@stefanurkal Please submit one and only one sample program per pull request (PR). You will need to create a different branch for each sample program.

rzuckerm commented on archive/c/coffeescript/longest-word.coffee:
```diff
@@ -0,0 +1,25 @@
+main = () ->
+    args = process.argv
+    return findLongestWord(args[2])
+    usage()
```
It doesn't seem like this line would ever be executed since it happens after `return`

rzuckerm commented on archive/c/coffeescript/longest-word.coffee:
```diff
@@ -0,0 +1,25 @@
+    
+main = () ->
+    args = process.argv
+    return findLongestWord(args[2])
```
Although I'm not familiar with this language, it seems like you'd need to check the length of `args` before accessing index 2.

rzuckerm commented on archive/r/red/fibonacci.red:
```diff
@@ -0,0 +1,24 @@
+Red [Title: "Fibonacci"]
```
This sample needs to be removed from the PR

stefanurkal commented:
Updated code, and removed the one line and previous code.  In coffee script Args always start at array value [2], other passed examples also have this in their code for this language

rzuckerm reviewed (approved):
Looks good, @stefanurkal . Thanks for your contribution!
